src/InterfaceSymbolicUtils.jl

```diff
@@ -5,225 +5,132 @@ using SymbolicUtils
+function Base.convert(::typeof(Node), x::Symbol, options::Options; varMap::Union{Array{String, 1}, Nothing}=nothing)
+    varMap == nothing && return Node(String(x))
+    return Node(String(x), varMap=varMap)
 end
 
-function unaop_to_index(op::F, options::Options)::Int where {F<:Function}
-    for i=1:options.nuna
-        if op == options.unaops[i]
-            return i
-        end
+function Base.convert(::typeof(Node), x::SymbolicUtils.Symbolic, options::Options; varMap::Union{Array{String, 1}, Nothing}=nothing)
+    if !SymbolicUtils.istree(x)
+        varMap == nothing && return Node(String(x.name))
+        return Node(String(x.name), varMap)
     end
-    error("Operator $(op) in simplified expression not found in options $(options.unaops)!")
-end
 
-function binop_to_index(op::F, options::Options)::Int where {F<:Function}
-    for i=1:options.nbin
-        if op == options.binops[i]
-            return i
-        end
-    end
-    error("Operator $(op) in simplified expression not found in options $(options.binops)!")
+    op = convert_to_function(SymbolicUtils.operation(x))
+    args = SymbolicUtils.arguments(x)
+
+    length(args) > 2 && return split_eq(op, args, options; varMap=varMap)
+    ind = length(args) == 2 ? findoperation(op, options.binops) : findoperation(op, options.unaops)
+
+    return Node(ind, map(x->convert(Node, x, options, varMap=varMap), args)...)
 end
 
-function varMap_to_index(var::Symbol, varMap::Array{String, 1})::Int
-    str = string(var)
-    for i=1:length(varMap)
-        if str == varMap[i]
-            return i
-        end
-    end
+"""
+    node_to_symbolic(tree::Node, options::Options;
+                varMap::Union{Array{String, 1}, Nothing}=nothing,
+                evaluate_functions::Bool=false,
+                index_functions::Bool=false)
+
+The interface to SymbolicUtils.jl. Passing a tree to this function
+will generate a symbolic equation in SymbolicUtils.jl format.
+
+## Arguments
+
+- `tree::Node`: The equation to convert.
+- `options::Options`: Options, which contains the operators used in the equation.
+- `varMap::Union{Array{String, 1}, Nothing}=nothing`: What variable names to use for
+    each feature. Default is [x1, x2, x3, ...].
+- `evaluate_functions::Bool=false`: Whether to evaluate the operators, or
+    leave them as symbolic.
+- `index_functions::Bool=false`: Whether to generate special names for the
+    operators, which then allows one to convert back to a `Node` format
+    using `symbolic_to_node`.
+"""
+function node_to_symbolic(tree::Node, options::Options;
+                     varMap::Union{Array{String, 1}, Nothing}=nothing,
+                     evaluate_functions::Bool=false,
+                     index_functions::Bool=false
+                     )
+    expr = subs_bad(parse_tree_to_eqs(tree, options, index_functions, evaluate_functions))
+    # Check for NaN and Inf
+    @assert isgood(expr) "The recovered equation contains NaN or Inf."
```

Comment:
This is the reason for the try catch. We could just get rid of it and check the result for Inf / NaN here.

Reply:
I think it's fine as is!